From 15218341c1b36e23d9e492bb5ef7df65e2563edf Mon Sep 17 00:00:00 2001
From: Jonathon Jongsma <jjongsma@redhat.com>
Date: Fri, 17 Jul 2015 09:16:10 -0500
Subject: [PATCH] Set enabled status of all displays when we get a monitor
 update

Previously, when we received a new monitors update from the server, we
only called virt_viewer_display_set_enabled() for the displays that were
enabled. We simply assumed that those that were not enabled were already
set to disabled. This assumption is currently valid, but I have some
changes in the pipeline where this is not true. This change ensures that
we update the enabled state of all monitors when we get an updated
monitors conifguration.

(cherry picked from commit 344eb9a68595ee26627e27fe733ac7789fdbdbb1)
---
 src/virt-viewer-session-spice.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/virt-viewer-session-spice.c b/src/virt-viewer-session-spice.c
index d3d7edc..506c2a7 100644
--- a/src/virt-viewer-session-spice.c
+++ b/src/virt-viewer-session-spice.c
@@ -720,13 +720,15 @@ virt_viewer_session_spice_display_monitors(SpiceChannel *channel,
 
     for (i = 0; i < monitors->len; i++) {
         SpiceDisplayMonitorConfig *monitor = &g_array_index(monitors, SpiceDisplayMonitorConfig, i);
+        gboolean disabled = monitor->width == 0 || monitor->height == 0;
         display = g_ptr_array_index(displays, monitor->id);
         g_return_if_fail(display != NULL);
 
-        if (monitor->width == 0 || monitor->height == 0)
+        virt_viewer_display_set_enabled(VIRT_VIEWER_DISPLAY(display), !disabled);
+
+        if (disabled)
             continue;
 
-        virt_viewer_display_set_enabled(VIRT_VIEWER_DISPLAY(display), TRUE);
         virt_viewer_display_spice_set_desktop(VIRT_VIEWER_DISPLAY(display),
                                               monitor->x, monitor->y,
                                               monitor->width, monitor->height);
